lib/preview-web/src/PreviewWeb.test.ts

```diff
@@ -0,0 +1,2084 @@
+        expect(projectAnnotations.renderToDOM).toHaveBeenCalledTimes(1);
+        expect(projectAnnotations.renderToDOM).toHaveBeenCalledWith(
+          expect.objectContaining({
+            forceRemount: true,
+            storyContext: expect.objectContaining({
+              loaded: { l: 8 },
+              args: { foo: 'a', new: 'arg' },
+            }),
+          }),
+          undefined // this is coming from view.prepareForStory, not super important
+        );
+      });
+
+      it('renders a second time if renderToDOM is running', async () => {
+        const [gate, openGate] = createGate();
+
+        document.location.search = '?id=component-one--a';
+        projectAnnotations.renderToDOM.mockImplementationOnce(async () => gate);
+        await new PreviewWeb({ getProjectAnnotations, importFn, fetchStoryIndex }).initialize();
+
+        emitter.emit(Events.UPDATE_STORY_ARGS, {
+          storyId: 'component-one--a',
+          updatedArgs: { new: 'arg' },
+        });
+
+        // Now let the renderToDOM call resolve
+        openGate();
+        await waitForRender();
+
+        expect(projectAnnotations.renderToDOM).toHaveBeenCalledTimes(2);
+        expect(projectAnnotations.renderToDOM).toHaveBeenCalledWith(
+          expect.objectContaining({
+            forceRemount: true,
+            storyContext: expect.objectContaining({
+              loaded: { l: 7 },
+              args: { foo: 'a' },
+            }),
+          }),
+          undefined // this is coming from view.prepareForStory, not super important
+        );
+        expect(projectAnnotations.renderToDOM).toHaveBeenCalledWith(
+          expect.objectContaining({
+            forceRemount: false,
+            storyContext: expect.objectContaining({
+              loaded: { l: 7 },
+              args: { foo: 'a', new: 'arg' },
+            }),
+          }),
+          undefined // this is coming from view.prepareForStory, not super important
+        );
+      });
+
+      it('works if it is called directly from inside non async renderToDOM', async () => {
+        document.location.search = '?id=component-one--a';
+        projectAnnotations.renderToDOM.mockImplementationOnce(() => {
+          emitter.emit(Events.UPDATE_STORY_ARGS, {
+            storyId: 'component-one--a',
+            updatedArgs: { new: 'arg' },
+          });
+        });
+        await new PreviewWeb({ getProjectAnnotations, importFn, fetchStoryIndex }).initialize();
+
+        await waitForRender();
+
+        expect(projectAnnotations.renderToDOM).toHaveBeenCalledTimes(2);
+        expect(projectAnnotations.renderToDOM).toHaveBeenCalledWith(
+          expect.objectContaining({
+            forceRemount: true,
+            storyContext: expect.objectContaining({
+              loaded: { l: 7 },
+              args: { foo: 'a' },
+            }),
+          }),
+          undefined // this is coming from view.prepareForStory, not super important
```

Comment:
can we just remove this if unimportant?

Reply:
`toBeCalledWith()` expects all arguments to be matched